test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -769,4 +779,41 @@ protected String minimalIsInvalidRoutingPathErrorMessage(Mapper mapper) {
             + mapper.typeName()
             + "].";
     }
+
+    /** Does this field type support copy_to */
+    protected boolean supportsCopyTo() {
+        return true;
+    }
+
+    public final void testCopyTo() throws IOException {
+
+        DocumentMapper mapper = createDocumentMapper(mapping(b -> {
+            b.startObject("field");
+            minimalMapping(b);
+            b.field("copy_to", "copy_field");
+            b.endObject();
+            b.startObject("copy_field");
+            minimalMapping(b);
+            b.endObject();
+        }));
+
+        if (supportsCopyTo()) {
+            ParsedDocument doc = mapper.parse(source(this::writeCopyField));
+            IndexableField[] source = doc.rootDoc().getFields("field");
+            IndexableField[] copy = doc.rootDoc().getFields("copy_field");
+
+            String sourceToString = Arrays.toString(source);
+            String expected = sourceToString.replaceAll("<field:", "<copy_field:");
```

Comment:
Is this the reason why toString was added to some mappers in this PR previously? It's short but I wonder if this could be tested without relying on some toString implementation in that case.

Reply:
`IndexableField` is an interface and doesn't implement `equals()`; however, all the lucene implementations *do* implement `toString()`, and adding it to the few ES implementations that didn't have it made this reasonably simple to test for.